parts/linux/cloud-init/artifacts/ensure_imds_restriction.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+# Ensure IMDS restriction rule is correctly set on the node, this script will be registered as kubelet service's ExecStart
+# to ensure desired iptables rule will be ensured after the node is restarted.
+# Below 3 variables are passed from kubelet service's drop-in file.
+primaryNicIP="${PRIMARY_NIC_IP}"
```

Comment:
I'd also recommend handling cases where any of the required environment variables are empty/unset

Reply:
updated code.